syntax = "proto3";
package mpmq;

message ProduceRequest  { string topic = 1; bytes value = 2; }
message ProduceReply    { int64  offset = 1; }
message ConsumeRequest  { string topic = 1; string group = 2; int32 max_records = 3; }
message Record          { int64 offset = 1; int64 ts = 2; bytes value = 3; }
message ConsumeReply    { repeated Record records = 1; }

service QueueService {
  rpc Produce(ProduceRequest) returns (ProduceReply);
  rpc Consume(ConsumeRequest) returns (ConsumeReply);
}